waiter/src/waiter/core.clj

```diff
@@ -814,11 +814,11 @@
    :service-id->source-tokens-entries-fn (pc/fnk [[:curator kv-store]]
                                            (partial sd/service-id->source-tokens-entries kv-store))
    :start-new-service-fn (pc/fnk [[:scheduler scheduler]
-                                  [:state authenticator start-service-cache task-thread-pool]
+                                  [:state start-service-cache task-thread-pool]
                                   store-service-description-fn]
                            (fn start-new-service [{:keys [service-id] :as descriptor}]
                              (let [run-as-user (get-in descriptor [:service-description "run-as-user"])]
-                               (auth/check-user authenticator run-as-user service-id))
+                               (scheduler/validate-user scheduler run-as-user service-id))
```

Comment:
I would like this to have a more generic name and signature `(validate this service-id)`. We can avoid passing the `run-as-user` as the scheduler can retrieve it from the service description.